src/main/java/com/impactupgrade/nucleus/service/segment/SendGridEmailService.java

```diff
@@ -44,16 +45,18 @@ public void sendEmailTemplate(String subject, String template, Map<String, Objec
     request.setMethod(Method.POST);
     request.setEndpoint("/mail/send");
 
-    Mail mail = new Mail();
-
-    for (String to : tos) {
-      Personalization personalization = new Personalization();
-      personalization.addTo(new Email(to));
-      for (Map.Entry<String, Object> d : data.entrySet()) {
-        personalization.addDynamicTemplateData(d.getKey(), d.getValue());
-      }
-      mail.addPersonalization(personalization);
+    Personalization personalization = new Personalization();
+    tos.forEach(to -> personalization.addTo(new Email(to)));
+    if (CollectionUtils.isNotEmpty(ccs)) {
+      ccs.forEach(cc -> personalization.addCc(new Email(cc)));
+    }
+    if (CollectionUtils.isNotEmpty(bccs)) {
+      bccs.forEach(bcc -> personalization.addBcc(new Email(bcc)));
     }
+    data.entrySet().forEach(e -> personalization.addDynamicTemplateData(e.getKey(), e.getValue()));
+
+    Mail mail = new Mail();
+    mail.addPersonalization(personalization);
```

Comment:
According to the doc:
https://www.twilio.com/docs/sendgrid/for-developers/sending-email/personalizations#personalization-examples